[MMB-174] Remove named cursors (#91)
ably-labs/spaces: docs/class-definitions.md

## Methods

### set

> Set a `CursorUpdate`. If a member has not yet entered the space, this method will error.

Reviewer: ```suggestion
Set the position of a cursor. This will emit a `CursorUpdate` event. If a member has not yet entered the space, this method will error.
```

> `CursorUpdate` is an object with 2 properties. `position` is an object with 2 required properties, `x` and `y`. These represent the position of the cursor on a 2D plane. A second property, `data` can passed. This is an object of any shape and is meant for data associated with the cursor movement (like drag or hover calculation results):

Reviewer: ```suggestion
A `CursorUpdate` is an object with 2 properties. `position` is an object with 2 required properties, `x` and `y`. These represent the position of the cursor on a 2D plane. A second optional property, `data` can also be passed. This is an object of any shape and is meant for data associated with the cursor movement (like drag or hover calculation results):
```

```ts
type set = (update: { position: CursorPosition, data?: CursorData })
```

Example usage:

```ts
window.addEventListner('mousemove', ({ clientX, clientY }) => {
  space.cursors.set({ position: { x: clientX, y: clientY }, data: { color: "red" } });
});
```

### getAll

Get the last CursorUpdate for each connection.

```ts
type getAll = () => Record<ConnectionId, CursorUpdate>;
```

Example:

```ts
const lastPositions = space.cursors.getAll();
```

### on

> Listen to cursor update events. See [EventEmitter](/docs/usage.md#event-emitters) for overloading usage.

Reviewer: ```suggestion
Listen to `CursorUpdate` events. See [EventEmitter](/docs/usage.md#event-emitters) for overloading usage.
```
